implicits/src-3/upickle/implicits/CaseClassReader.scala

```diff
@@ -82,11 +99,27 @@ trait CaseClassReaderPiece extends MacrosCommon:
       Reader.merge[T](readers:_*)
   }
 
+  inline def macroEnumR[T](using m: Mirror.Of[T]): Reader[T] = inline m match {
+    case m: Mirror.ProductOf[T] =>
+      throw new UnsupportedOperationException("Generated Enum Writer should never encounter Product")
+
+    case m: Mirror.SumOf[T] =>
+      val valueOf = macros.enumValueOf[T]
+      val description = macros.enumDescription[T]
+      new EnumReader[T](valueOf, description)
+  }
```

Comment:
For backwards compatibility, I think it would be useful to have this method folded into in the `macroR` method.This could be merged as a case in the general `macroR` method. E.g.

```scala
inline def macroR[T](using m: Mirror.Of[T]): Reader[T] = inline m match {
  case m: Mirror.ProductOf[T] => ...
  case m: SumOf[T] if erasedValue[T].isInstanceOf[scala.reflect.Enum] => <content of macroEnumR>
  case m: SumOf[T] => ...
}
```

The reason I say that this is for backwards compatibility, is that there is lot of scala 2 code out there defines implicit ReadWriters calling to the macroR method, and it would be nice if we would get new enum functionality in those cases too, even if the old syntax is still kept around.